### ISSUE TITLE
QA of HPOA Ingest KGX

### ISSUE THREAD

**Issue by mbrush** [NCATSTranslator/translator-ingests#212]
**QA Resources:** 
- KGX Summary report: https://docs.google.com/spreadsheets/d/1Zr2oJ_iwDiWxpn-qEEp19eSLk5b4d0mAlKUUgOB690s/edit?gid=527165474#gid=527165474
- RIG: https://github.com/NCATSTranslator/translator-ingests/blob/main/src/translator_ingest/ingests/hpoa/hpoa_rig.yaml

**Original Phase 2 KGX inventory:**
- https://docs.google.com/spreadsheets/d/1iLpJ3vULH058rBxlGv_soUyZw107OORZ-cUOgo3Lz3M/edit?pli=1&gid=325339947#gid=325339947

**Issues to consider / address for January updated release:**
- Overall, the ingest code does not seem to be generating the qualified versions of the edge types as specified in the RIG.  
   - e.g. `causes` and `contributes` to are showing up as the primary predicate, rather than the qualified predicate where they should be, and SubjectFormOrVariant qualifiers are not showing up on these edges 
       - **_This concern resolved in patch to  existing PR #205_** 
- Use of `has_evidence` edge property.  Known issue in Biolink - see here (https://github.com/biolink/biolink-model/issues/1664) - this property is defined to take an 'instance' of evidence, but bound to an EvidenceType class which is defined to hold evidence codes (e.g. ECO terms).   Address Biolink issue, then update property used here. 
    - **_RB Note: No change needed to HPOA ingest, but Biolink Model PR was issued. See comment https://github.com/NCATSTranslator/translator-ingests/issues/212#issuecomment-3700236162 below_.**
- There is one Protein causes gene edge that is likely a result of an odd node normalization (row 52 in the examples)
    - **_RB Note: No change needed to HPOA ingest, but possible NN-related error should be investigated. See comment https://github.com/NCATSTranslator/translator-ingests/issues/212#issuecomment-3700292443 below._**

**Issues to consider longer term:**
- . . .

**Comment by RichardBruskiewich** [comment]
- Overall, the ingest code does not seem to be generating the qualified versions of the edge types as specified in the RIG.  
   - e.g. `causes` and `contributes` to are showing up as the primary predicate, rather than the qualified predicate where they should be, and SubjectFormOrVariant qualifiers are not showing up on these edges 
       - **_RB Note: please see the [original Phase 2 inventory of edges](https://docs.google.com/spreadsheets/d/1iLpJ3vULH058rBxlGv_soUyZw107OORZ-cUOgo3Lz3M/edit?pli=1&gid=325339947#gid=325339947).  There were qualifiers from HPOA but I don't see  SubjectFormOrVariant._**

> **QA Resources:**
> 
> * KGX Summary report: https://docs.google.com/spreadsheets/d/1Zr2oJ_iwDiWxpn-qEEp19eSLk5b4d0mAlKUUgOB690s/edit?gid=527165474#gid=527165474
> * RIG: https://github.com/NCATSTranslator/translator-ingests/blob/main/src/translator_ingest/ingests/hpoa/hpoa_rig.yaml
> 
> **Original Phase 2 KGX inventory:**
> 
> * https://docs.google.com/spreadsheets/d/1iLpJ3vULH058rBxlGv_soUyZw107OORZ-cUOgo3Lz3M/edit?pli=1&gid=325339947#gid=325339947
> 
> **Issues to consider / address for January updated release:**
> 
> * Overall, the ingest code does not seem to be generating the qualified versions of the edge types as specified in the RIG.
>   
>   * e.g. `causes` and `contributes` to are showing up as the primary predicate, rather than the qualified predicate where they should be, and SubjectFormOrVariant qualifiers are not showing up on these edges
> * Use of `has_evidence` edge property.  Known issue in Biolink - see here ([fix/clarify `has evidence` association slot definition and usage. biolink/biolink-model#1664](https://github.com/biolink/biolink-model/issues/1664)) - this property is defined to take an 'instance' of evidence, but bound to an EvidenceType class which is defined to hold evidence codes (e.g. ECO terms).   Address Biolink issue, then update property used here.
>   
>   * **_RB Note: No change needed to HPOA ingest, but Biolink Model PR was issued. See comment [QA of HPOA Ingest KGX #212 (comment)](https://github.com/NCATSTranslator/translator-ingests/issues/212#issuecomment-3700236162) below_.**
> * There is one Protein causes gene edge that is likely a result of an odd node normalization (row 52 in the examples)
>   
>   * **_RB Note: No change needed to HPOA ingest, but possible NN-related error should be investigated. See comment [QA of HPOA Ingest KGX #212 (comment)](https://github.com/NCATSTranslator/translator-ingests/issues/212#issuecomment-3700292443) below._**
> 
> **Issues to consider longer term:**
> 
> * . . .

**Comment by RichardBruskiewich** [comment]
_>     * [ ]  There are a significant number (~86000, totaling ~1/3 of all D2P edges) of Disease -has_phenotype-> Disease edges in the data, that I think all start out with HP terms as objects, that get normalized to MONDO terms by babel/node norm.  e.g.   'nephrotic syndrome' HP:0000100 ->MONDO:0005377 (and see [these rows](https://docs.google.com/spreadsheets/d/1hChcsWzi5G37nEXqZQ2fH6jjcfADRLuhhrojuA1dQks/edit?gid=1436194094#gid=1436194094&range=17:21) for more examples). I am pointing this out in case we want to revisit the decision to make MONDO/Disease terms primary in all contexts when there is an equivalence between a disease and phenotype. I get that this is a necessary evil given the overlap in remit of MONDO and HP, and that consistently normalizing like this makes the data better linked and interoperable.  But here it may be confusing, as the predicate `has_phenotype` points to a Disease in over 1/3 of all d2P edges).  [@cmungall](https://github.com/cmungall) would love your advice here in particular.[ ]  Similarly, there are >108K Gene - has_phenotype -> Disease edges (compared to ~209K Gene - has_phenotype-> Phenotype).  Again, I would expect that in the source data, all objects of these annotations are Phenotype terms from HPOA. It seems strange that so many (108K) get tuned into Diseases (through Node Norm I assume?)._

Well, even just one example (below) from your spreadsheet @mbrush (cc: @cmungall), the existing conflation between HP ("phenotype") and MONDO ("diseases").  I suspect that if the Node Normalization ("NN") is absolutely biased towards a "preferred" namespace which lists MONDO ahead of HPO, then this semantic anomaly will persist.

_@cbizon (cc: @EvanDietzMorris), perhaps the preference for namespaces in NN could or should perhaps be tweaked to be more sensitive to the existing association ("edge") **category** context of the statement being normalized, to the extent that (@sierra-moxon ?) the Biolink Model accordingly constrains the categorical **range** of the subject and object of the specific association?_

## Sample anomalous definitions 

(As seen in the NN'd HPOA documented in line 12 of [Matt's sample output](https://docs.google.com/spreadsheets/d/1hChcsWzi5G37nEXqZQ2fH6jjcfADRLuhhrojuA1dQks/edit?gid=1436194094#gid=1436194094&range=17:21) above)

### From HPO:
* Phenotypic abnormality (4,751)
    * Abnormality of blood and blood-forming tissues (191)
        * Abnormality of thrombocytes (9)
            * Abnormal platelet morphology (7)
                * Abnormal platelet count (2)
                    * Thrombocytopenia

### From MONDO:
* disease (25,947)
    * human disease (22,986)
        * hematologic disorder (1,313)
            * blood platelet disease (93)
                * thrombocytopenia (56)

### Term definition from MalaCards:
[Thrombocytopenia](https://www.malacards.org/card/thrombocytopenia): is a blood platelet disease characterized by abnormally low numbers of platelets (thrombocytes) in the blood. Platelets are produced in bone marrow and travel through blood vessels to stick together and form clots (thrombi) that stop bleeding. Reduced platelet levels can cause prolonged or excessive bleeding, with clinical signs including increased bruising or hemorrhaging. Diagnosis is confirmed by identifying a decreased platelet count in a blood sample. Thrombocytopenia is the most common coagulation disorder among intensive care patients and occurs in about one-fifth of medical patients and one-third of surgical patients.

**Comment by sandrine-muller-research** [comment]
Thank you for bringing up these questions in this ticket. I do not want to derail the discussion but rather would like to bring representation issues while working on our FSHD1 use case with colleagues as a way to bring a use case to help reflect on this particular question:
_I wanted to confirm the that for G2D associations, when the "association_type" field = 'UNKNOWN' we want to use the more general 'associated with' predicate, as opposed to 'causes or contributes to'. In these cases, has a causal/contributing relationship has not been established - i.e. the relationship could be something else (a correlation). Or is it just that we don't know if the gene is causal on its own vs one of many possible contributing factors (in which case causes_or_contributes_to would be appropriate)?_

## Background:
1.  FSHD1 is a rare disease with causal (in the genetics signification) sequences variants (at the D4Z4 locus : 4q35, 4:190000-4:191000).
2.  D4Z4 variants are causal for FSHD1 **but not sufficient alone**; they require a permissive 4qA haplotype to enable DUX4 expression, **making the combination the direct disease trigger** 
3. In FSHD1, `D4Z4 contraction` **AND** `4qA haplotype's polyA signal ` cause the disease. FSHD1 **requires two hits** at the D4Z4 locus—(1) repeat contraction (<11 units), (2) permissive 4qA allele—mapping to one outcome (DUX4 toxicity). Neither **alone** causes disease.
4. This is the mechanistic explanation of the disease cause as decribed in the litterature:
```D4Z4_contraction [Variant, 4qA_haplotype qualifier]
     ↓ disrupts(chromatin_state)
heterochromatin [ChromatinState]
     ↓ negatively_regulates(regulates_expression)  
DUX4 [Gene, polyA_stabilized_mRNA qualifier]
     ↓ causes(pathology)
FSHD [Disease]
```

## Reasoning towards the modelization using this use case:
A factor is causal (from a genetics point of view) if altering it changes disease probability (e.g., D4Z4 contraction → chromatin relaxation → DUX4 misexpression → FSHD). It's **not sufficient** without the 4qA haplotype's polyA signal stabilizing DUX4 mRNA; ~20% of 4q chromosomes lack this, blocking toxicity despite contraction. Contractions on non-permissive 4qB haplotypes are harmless.

### 1. Modeling the relationship as a simple `biolink:causes` predicate :
The triple implies direct, monocausal sufficiency, **ignoring the necessary 4qA haplotype co-occurrence** (~80% of 4q chromosomes carry it, but only contractions on 4qA trigger disease). It flattens the component cause model into a single edge, blocking queries like "What fraction of D4Z4 contractions are pathogenic?" (answer: ~80%, haplotype-dependent) or "Does 4qB contraction cause FSHD?" (no).
**Modeling "D4Z4 contraction causes FSHD1" as a simple triple (e.g., D4Z4_variant --> causes --> FSHD1) captures a _genetic truth_ but loses the _plurifactorial reality_, making it incomplete for robust reasoning or KG inference IMO.**

### 2. Modeling the relationship as a qualified predicate:
As my attempt (I may not be up to date with the qualifiers representation) the accurate qualified modelization should become (let's ignore the fact that we cannot represent this appropriately to focus on the reasoning):
```
associations:
  - id: assoc1
    type: GenotypeToDiseaseAssociation
    subject: D4Z4_contraction_4qA
    predicate: associated_with 
    qualified_predicate: contributes_to
    object: FSHD1
    qualifiers:
      - qualifier_type: haplotype
        qualifier_value: 4qA
      - qualifier_type: repeat_count 
        qualifier_value: "<11"
```
While qualifiers allow to filter/specify rapidly a triple, I view Biolink qualifiers as a "metagraph" embedding : they create a nested subgraph "structure" where complexity (in our case: haplotype dependencies) lives as **interconnected nodes within the qualifier array**, enabling layered reasoning without bloating the core triple.

Qualified predicates refine one association between a subject and object with qualifiers (e.g., `D4Z4_contraction` `qualified_predicate: disrupts heterochromatin` with `object_aspect: chromatin_state`)

**My understanding, when modeling MolePro qualified predicates, was to be able to represent associations between two entities _where the full mechanism is unknwown/not represented by the source_.** 


### 3. Using HPO - another layer of complexity: 
The only G2D relationship on HPO for FSHD1 is _interestingly_ [FRG1](https://hpo.jax.org/browse/disease/OMIM:158900). From the description provided:
```
FRG1 NCBIGene:2483
Gene Location: 4q35.2

Definition

This gene maps to a location 100 kb centromeric of the repeat units on chromosome 4q35 which are deleted in facioscapulohumeral muscular dystrophy (FSHD). It is evolutionarily conserved and has related sequences on multiple human chromosomes but DNA sequence analysis did not reveal any homology to known genes. In vivo studies demonstrate the encoded protein is localized to the nucleolus. [provided by RefSeq, Jul 2008]
```
 I (from my understanding) would model that generating 3 associations:
```
associations:
  - type: GeneToDiseaseAssociation
    subject: FRG1
    predicate: associated_with
    qualified_predicate: located_near
    object: FSHD1
    qualifiers:
      - qualifier_type_id: biolink:proximity
        qualifier_value: "100kb_centromeric_to_D4Z4"

associations:
  - type: GeneToGenomicLocusAssociation  
    subject: FRG1
    predicate: located_in
    qualified_predicate: adjacent_to
    object: D4Z4_repeat_array
    qualifiers:
      - qualifier_type_id: biolink:distance
        qualifier_value: "100kb"
      - qualifier_type_id: biolink:spatial_orientation  
        qualifier_value: "centromeric"

associations:
  - id: assoc_hpo_inferred
    type: GeneToDiseaseAssociation
    subject: FRG1
    predicate: associated_with
    qualified_predicate: related_to ???? is that possible with biolink?
    object: FSHD1
    qualifiers:
      - object_aspect_qualifier: location
      - publications: ["OMIM:158900"]
      - original_knowledge_source: "hpo.jax.org"

```
The third statement would read: _FRG1 is related to FSHD1 in location aspect from HPO/OMIM_
- Captures positional association via 4q35.2 proximity
- No causal claim—matches evidence
- Preserves HPO annotation

`contributes_to` would falsely read: "FRG1 contributes to FSHD1"
- Wrongly implies FRG1 drives pathology
- apart from HPO: contradicts consensus (DUX4 is causal)

When I look at [FSHD genes to disease HPO associations](https://hpo.jax.org/browse/disease/ORPHA:269), I the folowing mapping pattern (not explicitely modeled):
 
```
associations:
  # 1. CAUSAL - FRG1
  - type: SequenceVariantToDiseaseAssociation
    subject: D4Z4
    predicate: associated_with
    qualified_predicate: causes
    object: FSHD1
    
  # 2. POSITIONAL - FRG1
  - type: GeneToDiseaseAssociation
    subject: FRG1
    predicate: associated_with
    qualified_predicate: related_to
    object: FSHD1
    qualifiers:
      - object_aspect_qualifier: location
      
  # 3. MODIFIER - SMCHD1 (FSHD2 mechanism)  
  - type: GeneToDiseaseAssociation
    subject: SMCHD1
    predicate: associated_with
    qualified_predicate: modifies
    object: FSHD1
    qualifiers:
      - object_aspect_qualifier: severity
      
  # 4. GENOMIC CONTEXT - FRG1 position
  - type: GeneToGenomicLocusAssociation
    subject: FRG1
    predicate: located_in
    qualified_predicate: adjacent_to
    object: D4Z4_repeat_array

```
If not explicitely modeled then the only mapping possibility would be IMO:
```
associations:
  - id: assoc_hpo_inferred
    type: GeneToDiseaseAssociation
    subject: FRG1
    predicate: [biolink:associated_with]
    qualified_predicate: related_to
    object: [biolink:Disease]
```
If we put `contributes_to` as a qualifier that means `contibutes_to`  maps to the follwing meanings:

<html>
<body>


HPO Association Type | qualified_predicate | description
-- | -- | --
Causal | causes | Direct Mendelian causality (stronger than contribution)
Positional | related_to | Purely spatial (FRG1), no causal mechanism
Modifier | modifies | Alters severity/course, not primary cause
Risk Factor | contributes_to_risk_of | Only this uses contribution semantics


</body>
</html>

**Comment by RichardBruskiewich** [comment]
84 HPOA output edges are tagged as **Gene--has_phenotype->PhenotypicFeature**.

We initially thought HPOA was missing data, but in fact, **Gene->PhenotypicFeature** edges are NOT **`has_phenotype`** predicate edges, but rather **`associated_with/causes`** edges, so in fact, what we are seeing is another issue: abnormal normalization of OMIM id's into genes rather than disease, for the observed edges

The number is tractably small:

> OMIM:188400 -> NCBIGene:1714 - DGCR DiGeorge syndrome chromosome region [ Homo sapiens (human)
> OMIM:151430 -> NCBIGene:596 - BCL2 BCL2 apoptosis regulator [ Homo sapiens (human) ]
> OMIM:111700 -> NCBIGene:6006 - RHCE Rh blood group CcEe antigens [ Homo sapiens (human) ]
> OMIM:142000 -> NCBIGene:3045 - HBD hemoglobin subunit delta [ Homo sapiens (human) ]
> OMIM:603855 -> NCBIGene:10167 - CFM1 cystic fibrosis modifier 1 [ Homo sapiens (human) ]
> OMIM:168820 -> NCBIGene:5444 - PON1 paraoxonase 1 [ Homo sapiens (human) ]
> OMIM:116790 -> NCBIGene:1312 - COMT catechol-O-methyltransferase [ Homo sapiens (human) ][3:13 PM]In fact, the first entry - DGCR - accounted for most of the published edges

We're not sure about the best way to 'fix' this, since all of the exceptions are in some sense, 'gene loci'. I think in some cases, certain variants of these gene loci do cause disease(?). And, in a sense, the DGCR gene locus is directly associated with a disease (DiGeorge)
